

# Project set-up


## Directories

Scripts are in scripts, analysis contains mostly files generated by tools such as multiqc, data contains raw, intermediate data and counts.


## Steps

The steps of the pipeline are as follow :

1.  Optional :
    Download SRA data with `prefetch` and decompress with `fastq-dump`
2.  Quality control with `fastqc` and `multiqc`<sup><a id="fnr.1" class="footref" href="#fn.1">1</a></sup>
3.  Adapter trimming with `cutadapt`<sup><a id="fnr.2" class="footref" href="#fn.2">2</a></sup>
4.  Pseudo-mapping and quantification with `salmon`<sup><a id="fnr.3" class="footref" href="#fn.3">3</a></sup>
5.  Importation/transcript quantification are done with `tximeta`<sup><a id="fnr.4" class="footref" href="#fn.4">4</a></sup>
6.  DEG analysis with `DESeq2`<sup><a id="fnr.5" class="footref" href="#fn.5">5</a></sup>
7.  Clustering analysis with `WGCNA`<sup><a id="fnr.6" class="footref" href="#fn.6">6</a></sup> and `GWENA`<sup><a id="fnr.7" class="footref" href="#fn.7">7</a></sup>


## Workflow tools

-   steps 2-3-4 are performed via `Snakemake`<sup><a id="fnr.8" class="footref" href="#fn.8">8</a></sup>, using also `refgenie`<sup><a id="fnr.9" class="footref" href="#fn.9">9</a></sup>
-   steps 5-6-7 through a `shiny` interface


# Usage


## From raw data (reads)

1.  [Clone](https://www.git-scm.com/docs/git-clone) this repository (`git clone https://github.com/paulimer/RNA-seq-analysis-pipeline`)
2.  Create a directory `seqs` under `data` (path relative to project root : `data/seqs`)
3.  Put your fastq files there (these can be compressed files)
4.  Alternatively, download files from SRA with the `download_dump.py` script.
    For that follow those steps :
    1.  Download the Accession list from the SRA, and save it under `SRR_Acc_List.txt` in the project root
    2.  **In progress** there are options for single and paired-end download
    3.  Run `python scripts/download_dump.py` from the project root
5.  Install [Mamba](https://mamba.readthedocs.io/en/latest/installation.html)
6.  Run `mamba env create -f salmon_env.yml` in a terminal
7.  Run `conda activate name_of_env`
8.  Edit the config file
    For instance with the file containing the names of the samples
9.  Run `snakemake --snakefile <snakefile-chosen> --cores number_of_cores_chosen`
    You can choose between to snakefile, one for single end (`snakefile_single`) and one for paired end data (`snakefile_paired`).
    The minimum number of cores is two, because that's to pipe cutadapt into salmon and reduce disk usage
10. Quality control check in `./analysis`
11. Run `R -e "shiny::runApp('/scripts/shiny')"` for the rest of the analysis


## From a raw counts table

1.  [Clone](https://www.git-scm.com/docs/git-clone) this repository (`git clone https://github.com/paulimer/RNA-seq-analysis-pipeline`)
2.  Create a sample data table, with a *names* colum containing the sample names as specified in the counts table, and additional variables, such as this :

<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">names</th>
<th scope="col" class="org-left">condition</th>
</tr>
</thead>

<tbody>
<tr>
<td class="org-left">SRR7798788</td>
<td class="org-left">vehicle</td>
</tr>


<tr>
<td class="org-left">SRR7798789</td>
<td class="org-left">PDGF\<sub>BB</sub></td>
</tr>


<tr>
<td class="org-left">SRR7798790</td>
<td class="org-left">TGF\<sub>beta</sub></td>
</tr>


<tr>
<td class="org-left">SRR7798791</td>
<td class="org-left">vehicle</td>
</tr>


<tr>
<td class="org-left">SRR7798792</td>
<td class="org-left">PDGF\<sub>BB</sub></td>
</tr>


<tr>
<td class="org-left">SRR7798793</td>
<td class="org-left">TGF\<sub>beta</sub></td>
</tr>


<tr>
<td class="org-left">SRR7798794</td>
<td class="org-left">vehicle</td>
</tr>


<tr>
<td class="org-left">SRR7798795</td>
<td class="org-left">PDGF\<sub>BB</sub></td>
</tr>


<tr>
<td class="org-left">SRR7798796</td>
<td class="org-left">TGF\<sub>beta</sub></td>
</tr>
</tbody>
</table>

Ideally, one should not use symbols such as `-`, `/` and spaces in the names or other column of this sample data table.

1.  Run `R -e "shiny::runApp('/scripts/shiny')"`


# Footnotes

<sup><a id="fn.1" href="#fnr.1">1</a></sup> <https://doi.org/10.1093/bioinformatics/btw354>

<sup><a id="fn.2" href="#fnr.2">2</a></sup> <https://doi.org/10.14806/ej.17.1.200>

<sup><a id="fn.3" href="#fnr.3">3</a></sup> <https://doi.org/10.1038/nmeth.4197>

<sup><a id="fn.4" href="#fnr.4">4</a></sup> <https://doi.org/10.1371/journal.pcbi.1007664>

<sup><a id="fn.5" href="#fnr.5">5</a></sup> <https://doi.org/10.1186/s13059-014-0550-8>

<sup><a id="fn.6" href="#fnr.6">6</a></sup> <https://doi.org/10.1186/1471-2105-9-559>

<sup><a id="fn.7" href="#fnr.7">7</a></sup> <https://doi.org/10.1186/s12859-021-04179-4>

<sup><a id="fn.8" href="#fnr.8">8</a></sup> <https://doi.org/10.1093/bioinformatics/bts480>

<sup><a id="fn.9" href="#fnr.9">9</a></sup> <https://doi.org/10.1093/gigascience/giz149>
