* TODO Goals
** TODO R DESeq2 scripts
*** TODO Understanding [4/8]
- [ ] Why use padj when there is FDR? Is padj any good?
- [ ] =filterThreshold=? What is it? Independent filtering?
- [X] Why take the 20 000 most up/down-regulated genes? Why 20k? Not adaptative to dataset.
- [X] Why plot again in the analysis (especially =plotMA=?)
- [ ] What are the bins for (to understand, not to correct)
- [X] What is rlog transformation file, and why use a log2(1+count) in there, as it is just a worse rlog? It is for a comparison it seems
- [X] Why sometimes =pheatmap=, sometimes =heatmap.2=?
- [ ] Investigate difference between GENCODE and Ensemble gene names, regarding issues with "duplicate genes on haplotype chromosomes"

*** DONE Correction [5/5] [2021-08-04 Wed]
- [X] Clean up and comment Soham's code
- [X] Maybe use built-in =Maplot= and =pcaPlot= functions? Same for =volcanoplot=?
- [X] Why export =volcanoplot= to postcript aaaah
- [X] Where are the =$treatment= and =$patient= coming from?
- [X] Is there no better standard way to plot distance between samples? see MSMB

*** DONE Adaptation [4/4]  [2021-08-04 Wed]
- [X] Interface the beginning with =salmon= via =tximeta= in order to annotate and de-normalize the data for =DESeq2=.
- [X] Make all the plots export to svg/png in a dedicated analysis subfolder, for further analysis/evaluation
- [X] On the same note, do the same with the =.csv= output, but maybe in data?
- [X] Install relevant packages

 
** TODO R WGCNA scripts [8/11]
- [X] Clean up Soham's code
- [X] Run GWENA on some samples? Or do only WGCNA?
- [X] Figure out what at which point to get normalized counts (and nothing else) from my pipeline :
  The problem is that DESeq2 takes un-normalized counts, salmon provides normalized counts : =counts(dds, normalized = TRUE)= provides DESeq2-normalized counts.
- [X] Integrate the basic analysis into shiny
- [X] Compare functional enrichment across %tages of filtered genes
- [X] A subset of genes is selected (else it is too resource-intensive), the highest-variance ones. However, we've seen that there is a mean-variance relationship in count data. So our selection is biased towards highly expressed genes (is that bad?), unless we use a variance stabilizing transformation.
- [X] Pre-screen data to eliminate outlier samples!
- [X] Put a warning on sample count
- [ ] Look at differential co-expression analysis
- [ ] Try to parallelize? Or make a note for someone to do it afterwards. Or try iit once you have 16GB of RAM.
- [ ] Display number of modules?



** TODO shiny [18/24]
- [X] Choose the experimental design (add variables for instance)
- [X] Choose what condition is control
- [X] logFoldChange and p-value cutoffs (on sliders?)
- [X] Show the end of computation of all the =DESeq= functions. (not needed?)
- [X] Top DE genes 
  - [X] numeric input
  - [X] renderDataTable and DataTableOutput from DT instead of plain shiny : it comes with a search function!
- [X] Multiple tabs, *exploratory analysis*, *table*, *Network analysis*
- [X] Allow download of table of genes, with =downloadButton()= (UI) and =downloadHandler()= (server)
- [X] Show number of up/downregultaed genes at current thresholds
- [X] Show gene names (after an annotation)
- [X] Plot particular genes (could be hard) could use function =selectize=, but it could be slow
- [X] Add selection of enrichment sources
- [X] Show only potential variables (not names and files)
- [X] Allow choosing of type of network
- [X] Check that pickstfthreshold only updates after a button prompt, but not the change of percent_
- [X] Running WGCNA without exploring samples bricks the app
- [ ] Plotting :
  - [ ] Draw a MAplot, add possibility to choose shrunken fold changes
  - [ ] Choose what to highlight in the PCA =intgroups=, maybe is only interesting if multiple variables
  - [ ] Allow brush to select some genes on the Maplot? Useful?
  - [ ] Add a  =fileInput= to enter the coldata information?
  - [ ] See that the figures scale better
  - [ ] Sample to sample distance does not plot?
- [ ] Look at annotating with tximport, and using GENCODE instead of ENSEMBL
- [ ] Barplot for gene counts with error bar(optional)?
- [X] Use shinydashboard to make things look pretty
- [X] Make ~my_values$res~ from an ~eventReactive~
- [ ] Add a shinyvalidate on the base/compare conditions, they need to be different
- [X] Check the c(input, input, value) are okay relative to logic.
- [ ] Make the max number of genes non-hardcoded
- [ ] Add possibility to start with counts table directly, instead of Salmon quants 

**  TODO Other scripts [1/2]
- [ ] Add an automatic download of the index with =refgenie= with the snakefile
- [X] Add refgenie to the mamba environment

  
** TODO General [0/5]
- [ ] Cite all papers related to packages used : credit where credit is due
- [ ] Update the README with steps
- [ ] Add a graph with the steps and the entry points
- [ ] Nettoyer le git repo (todos, etc)
- [ ] Ã©crire des scripts pour lancer :
  - [ ] Snakemake
  - [ ] shiny
 
 
